sql/src/main/java/io/crate/planner/operators/ProjectSet.java

```diff
@@ -57,15 +58,9 @@
             allUsedColumns.addAll(columnsUsedInTableFunctions);
             LogicalPlan sourcePlan = source.build(tableStats, hints, allUsedColumns, params);
 
-            // Use sourcePlan.outputs() as standalone to simply pass along all source outputs as well;
-            // Parent operators will discard them if not required
-            // The reason to do this is that we've no good way to detect what is required. E.g.
-            // select tableFunction(agg), agg, x
-            //  -> agg is used as argument in tableFunction, but is also standalone,
-            //     so we can't simply discard any source outputs that are used as arguments for the table functions.
-            //  -> x might be converted to _fetch by the Collect operator,
-            //       so we don't necessarily "get" the outputs we would expect based on the select list.
-            return new ProjectSet(sourcePlan, tableFunctions, sourcePlan.outputs());
+            ArrayList<Symbol> standalone = new ArrayList<>(sourcePlan.outputs());
+            standalone.removeAll(tableFunctions);
+            return new ProjectSet(sourcePlan, tableFunctions, standalone);
```

Comment:
Do you think it makes sense to port this change back to master as well? I think the column-pruning optimization "fixes" this in master.

Reply:
Yes makes sense.

Reply:
Had a look, it's not the column pruning but also that we build the operators differently. `sourcePlan.outputs()` will never contain the table functions in the first place.